crates/rust-analyzer/src/main_loop/handlers.rs

```diff
@@ -705,15 +704,9 @@ fn create_single_code_action(assist: Assist, world: &WorldSnapshot) -> Result<Co
         arguments: Some(vec![arg]),
     };
 
-    let kind = match assist.id {
-        AssistId("introduce_variable") => Some("refactor.extract.variable".to_string()),
-        AssistId("add_custom_impl") => Some("refactor.rewrite.add_custom_impl".to_string()),
-        _ => None,
-    };
-
     Ok(CodeAction {
         title,
-        kind,
+        kind: Some("refactor".to_owned()),
```

Comment:
it seems like using `Empty` by default, unless the kind is set explicitelly, is better?

Reply:
Defaulting to `Empty` makes sense. I just wanted things to show up in the Refactor menu but was too lazy to go through the work of restructuring everything.

Reply:
I now default to `Some(String::new)` which I think is fine until we start assigning kinds.